Assets/Scripts/Interactables/Dialogue/NPCDialogue/NPCInteractable.cs

```diff
@@ -1,62 +1,114 @@
-﻿using System.Collections;
-using System.Collections.Generic;
-using UnityEngine;
-
-[System.Serializable]
-public class MissionNPC
-{
-    public GameObject missionPrefab;
-
-    // Type of conversations an NPC can have based on if it is before, during or after a
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.AI;
+
+[System.Serializable]
+public class NPCMissionConvos
+{
+    public GameObject missionPrefab;
+
+    // Type of conversations an NPC can have based on if it is before, during or after a
     public Conversation beforeConvo;
     public Conversation duringConvo;
     public Conversation afterConvo;
-}
-
-public class NPCInteractable : DialogueInteractable
-{
-    public List<Conversation> defaultConvos;
-
-    public List<MissionNPC> missionsOffered;
-    private MissionNPC currentMissionNPC;                            // Current Mission given by this NPC (should only be 1 per NPC)
-    private AMission startedMission;                                 // started mission needed to end mission
-
-
-    public override void OnInteract()
+}
+
+public class NPCInteractable : DialogueInteractable
+{
+    public List<Conversation> defaultConvos;
+
+    public List<NPCMissionConvos> missionsOffered;
+    private NPCMissionConvos currentMissionConvos = null;            // Current Mission given by this NPC (should only be 1 per NPC)
+    private AMission startedMission;                                 // started mission needed to end mission
+
+
+    protected NavMeshAgent agent;
+    protected GameObject targetObject;
+
+    protected bool isFollowing = false;
+    private Vector3 originPosition;
+
+
+    protected override void Start()
     {
-        if (!isConversing)
+        base.Start();
+        agent = Utils.GetRequiredComponent<NavMeshAgent>(this);
+        originPosition = gameObject.transform.position;
+        LoadConversation();
+    }
+
+    protected override void Update()
+    {
+        if(IsWithinBoundaryRadius(GameManager.Instance.GetPlayerTransform()))
```

Comment:
Should we use like `targetObject` or something instead? If you wanted to keep some abstraction there? (I forget if tragetObject is the right variable or not though)

Reply:
I thought about that too but `IsWithinBoundaryRadius` is used for loading the conversation and determining how the NPC will interact with the player so it would have to know where player is relative to the NPC. Was thinking of using something like `targetObject` but realized we would have to first figure out a way to assign it as player. Couldn't really think of an efficient and easy way to do it off the bat so I just did this for now. Open to any suggestions or we can talk about it!

Reply:
Actually there may be a way looool. I'll try it out!